### PR TITLE
Workspace scroll bugfix and unit tests!

### PR DESCRIPTION
**Pull request by kchadha**
### Resolves

`Workspace scroll is wonky if block comments are on the screen. Scroll gets pinned to a corner of the workspace.` section of #1554.

### Proposed Changes

ScratchBlockComments should be able to handle `NaN` values for x and y positions, and their x and y positions properties should be updated after auto-positioning the comment (even if they are not rendered).

### Reason for Changes

The workspace metrics get wonky if scratch block comments are not correctly reporting positioning information.

### Test Coverage

Added unit tests! Particularly, `test_blockCommentXYNaNPositionProvided` and `test_blockCommentXYNullPositionProvided` fail prior to this fix and pass post changes.

### CODE REVIEW

File: core/scratch_block_comment.js [hunk 1]
```diff
@@ -58,17 +58,27 @@ Blockly.ScratchBlockComment = function(block, text, id, x, y, minimized) {
    */
   this.text_ = text;
   /**
-   * The x position of this comment in workspace coordinates.
+   * Whether this comment needs to be auto-positioned (based on provided values
+   * for x and y position).
+   * @type {boolean}
+   * @private
+   */
+  this.needsAutoPositioning_ = !x && x != 0 && !y && y != 0;
```
Comment (paulkaplan) [inline core/scratch_block_comment.js]: Is there a value in particular that causes this? Is it always null or NaN? this is a pretty hard to parse line, could be simplified if you knew it was always null/NaN and just checked for that. But not sure if there is really a better way.

File: core/scratch_block_comment.js [hunk 2]
```diff
@@ -58,17 +58,27 @@ Blockly.ScratchBlockComment = function(block, text, id, x, y, minimized) {
...
+  this.x_ = typeof x == 'number' && !isNaN(x) ? x : 0;
   /**
-   * The y position of this comment in workspace coordinates.
+   * The y position of this comment in workspace coordinates. Default to 0 if
+   * y position is not provided or is not a valid number.
    * @type {number}
    * @private
    */
-  this.y_ = y;
+  this.y_ = typeof y == 'number' && !isNaN(y) ? y : 0;
```
Comment (paulkaplan) [inline core/scratch_block_comment.js]: Could you do this check on x/y above and reuse the result of `typeof y == 'number' && !isNaN(y)` to make the line I commented on above more clear?

**Review by paulkaplan — APPROVED** [review]
I think this LG! I think @rachel-fenichel might have some other comments, but this feels good enough to go in.

**Comment by rachel-fenichel** [comment]
Yeah, LG.

File: core/scratch_block_comment.js (hunk 1 above)
Comment (kchadha) [reply core/scratch_block_comment.js]: I want the flag to be set if both x and y are any of: NaN, null, or undefined

**Comment by kchadha** [comment]
Woo! I'll wait for the tests and then merge. =D
